httpx/_transports/base.py

```diff
@@ -0,0 +1,129 @@
+import typing
+from types import TracebackType
+
+T = typing.TypeVar("T", bound="BaseTransport")
+A = typing.TypeVar("A", bound="AsyncBaseTransport")
+
+
+class BaseTransport:
+    def __enter__(self: T) -> T:
+        return self
+
+    def __exit__(
+        self,
+        exc_type: typing.Type[BaseException] = None,
+        exc_value: BaseException = None,
+        traceback: TracebackType = None,
+    ) -> None:
+        self.close()
+
+    def handle_request(
+        self,
+        method: bytes,
+        url: typing.Tuple[bytes, bytes, typing.Optional[int], bytes],
+        headers: typing.List[typing.Tuple[bytes, bytes]],
+        stream: typing.Iterator[bytes],
```

Comment:
I understand this used to be an `httpcore.SyncByteStream` type, and we're now switching to standard "can be used in a `for` loop" type.

Wondering if we should be using `Iterable[bytes]` instead?

Mostly because of this…

```python
def f(stream: Iterator[bytes]) -> None:
    pass

stream = [b"hello, world"]

f(stream)  # error: Argument 1 to "f" has incompatible type "List[bytes]"; expected "Iterator[bytes]"
```

And…

```python
def f() -> typing.Iterator[bytes]:
    return [b"content"]  # error: Incompatible return value type (got "List[bytes]", expected "Iterator[bytes]"
```

So:

```suggestion
        stream: typing.Iterable[bytes],
```

Reply:
Yup. So easy to get those two confused!